templates/FFV_form_factors.cpp.in

```diff
@@ -239,12 +365,214 @@ double OneLoopFunctionF(double r)
+                   * std::log(r)) / Power4(1.0 - r);
+   }
+}
+
+double OneLoopFunctionK(double r)
+{
+   if (is_zero(r)) {
+      return 1.0;
+   } else if (is_zero(1.0 - r)){
+      return 2.0 / 3.0;
+   } else if (1./r < Power3(ratiolimit)) {
+      // Loop function is about 10^-13, treat as reaching the 0 limit
+      return 0.;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return r*(-3.-2.*std::log(r));
+   } else {   
+      return (1.0 - 4.0 * r + 3.0 * r * r - 2 * r * r * std::log(r)) / Power3(1.0 - r);
+   }
+}
+
+double OneLoopFunctionL(double r)
+{  
+   if (is_zero(r)) {
+      return -10.0;
+   } else if (is_zero(1.0 - r)){
+      return 10.5;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return 25.*r;
+   } else {
+      return (2.0 + 27.0 * r - 54.0 * r * r + 25.0 * r * r * r - 6.0 * (2.0 - 9.0 * r + 6.0 * r * r)
+                 * std::log(r)) / Power4(1.0 - r);
+   }
+}
+
+double OneLoopFunctionM(double r)
+{
+   if (is_zero(r)) {
+      return 4.0;
+   } else if (is_zero(1.0 - r)){
+      return 1.5;
+   } else if (1./r < ratiolimit) {
+      // Expand about 0 to 1st order
+      return 5.*r;
+   } else {
+      return (4.0 - 9.0 * r + 5.0 * r * r * r + 6.0 * (1.0 - 2.0 * r) * r * std::log(r)) / 
+              Power4(1.0 - r);
    }
 }
 
+double OneLoopFunctionN(double r, double s)
+{
+   if (is_zero(r - s)) {
+      if (is_zero(r)) {
+         return 1.0;
+      } else if (is_zero(1.0 - r)){
+         return 2.0 / 3.0;
+      } else {
+         return (1.0 - 4.0 * r + 3 * r * r - 2 * r * r * std::log(r)) / 2.0 / Power3(1.0 - r);
+      }
+   } else if (1./s < ratiolimit) {
+      // Expand s about 0 to 1st order
+      return (1.0 - r * r * s + r * (s - 1.0) + r * (s + 1.0) * std::log(r)) / Power2(r - 1.0);
+   } else {
+      return s * (r * Power2(r - s) * std::log(r) + (r - 1.0) * ((r - s) * (s - 1.0) - (r - 1.0)
+                              * r * std::log(r/s))) / Power2(1.0 - r) / Power2(r - s) / (s - 1.0);
+   }
+}
+
+
+template<typename T>
+T KallenLambda(T m1, T m2, T m3)
+{
+   return Power2(m1-m2-m3) - 4*m2*m3;
+}
+
+double KallenLambdasqrt(double m1, double m2, double m3)
+{
+    return std::sqrt(KallenLambda(m1,m2,m3));
+}
+
+double DiscontinuityB(double m12, double m2, double m3)
+{
+    return KallenLambdasqrt(m12,Power2(m2),Power2(m3)) 
+	    * std::log((Power2(m2)+Power2(m3)-m12+KallenLambdasqrt(m12,Power2(m2),Power2(m3)))/(2.*m2*m3)) / m12;
+}
+
+double signed_dilog(double x, double a)
+{
+    if (a > 0.)
+    {
+        return -1./2. * Power2(std::log(1./(1.-x))) - dilog(x/(x-1.));
+    } else if (a < 0.) {
+        return dilog(x);
+    }
+}
+
+// function from eq. 2.9 of hep-ph/1105.4319
+// Standard C0 Passarino-Veltman function with first mass entry set to 0 (note the missing M12)
+double ScalarC0(double M22, double M32, double m1, double m2, double m3)
+{
+    double m12 = m12;
```

Comment:
These three lines are redundant. Does this even compile and give reasonable results?

Reply:
It did, strangely enough.  It is fixed now.

Reply:
Ok, thanks! But could you please make these new variables `const`? :)

Reply:
Ok, done.

Reply:
Thanks!